src/ContaoCoreBundle.php

```diff
@@ -57,6 +59,10 @@ public function build(ContainerBuilder $container): void
     {
         parent::build($container);
 
+        /** @var SecurityExtension $extension */
+        $extension = $container->getExtension('security');
```

Comment:
Is this the right place to add the factory or should we do it in a separate compiler pass?

Reply:
It's the same place Symfony does theirs (see https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/SecurityBundle/SecurityBundle.php#L45)